internal-rules/report-message-format.ts

```diff
@@ -0,0 +1,47 @@
+import type { TSESTree } from '@typescript-eslint/experimental-utils'
+import { ESLintUtils } from '@typescript-eslint/experimental-utils'
+import path from 'path'
+
+export const messageId = 'reportMessageFormat'
+type MessageIds = typeof messageId
+
+type Options = [string]
+
+export default ESLintUtils.RuleCreator(() => '')<Options, MessageIds>({
+  name: path.parse(__filename).name,
```

Comment:
I borrowed the rule name from https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/report-message-format.md, which makes perfect sense to them, since their rule also deals with `context.report({...})`... but for us, who only deal with `messages: { messageId }`, it might make more sense to call it ` message-id-format` or simply ` message format`.